BetterPasswordPlugin.inc.php

```diff
@@ -208,17 +199,109 @@ function checkPassword($hookName, $args) {
 			}
 		}
 	}
-
+        /**
+	 * Check for newly added blacklists. If ran the first time creates the list of blacklists
+	 * @return 0|1|2 0 if update failed, 1 if setting is updated, 2 if new setting has been created
+	 */
+        
+        function blacklistSetting() {
+            $prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+            $updateSettings = false;
+            $newBlacklist = array();
+            foreach ($this->getBlacklists() as $filename) {
+                    $newBlacklist[$filename] = sha1_file($filename);
+            }
+            if (is_null($prevBlacklist)) {
+                $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $newBlacklist);
+                $updateSettings = true;
+            }
+            else {
+                if ($prevBlacklist != $newBlacklist) {
+                    $updateTempFile = $this->handleTempFile(false);
+                    if($updateTempFile) {
+                        $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        $updateSettings = true;
+                    }
+                    else {
```

Comment:
Without additional code, this else with a standalone return only introduces ambiguity as to the function flow.

This would be a possible place to add an `error_log()` statement to describe that the `handleTempFile()` call failed, but that messaging would probably be better handled within the `handleTempFile()` method itself.  If there isn't logic to perform here, let the return from line 230 do the work.